src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,132 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
 
-    @Override
-    public void put(K key, V value) {
+    private int size;
+    private Node<K, V>[] buckets;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+    }
 
+    public void put(K key, V value) {
+        if (key == null) {
+            resize();
+            int index = 0;
+            Node<K, V> node = buckets[index];
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    node.value = value;
+                    return;
+                }
+                node = node.next;
+            }
+            buckets[index] = new Node<>(key, value, buckets[index]);
+            size++;
+        } else {
```

Comment:
I can't see reason to make if/else for null/notNull key.
Your getIndex method returns correct bucket index for null key (0), method Objects.equals will normally compare two null keys.
So whats unique in null key? It should normally behave like any other one